src/util/file_io.c

```diff
@@ -162,6 +162,18 @@ int read_bytes_from_file(char *input_path, uint8_t ** data,
   }
   int input_size = st.st_size;
 
+  if (input_size == 0)
+  {
+    kmyth_log(LOG_DEBUG, "input file size is zero, returning NULL ... exiting");
```

Comment:
It isn't necessarily expected to exit because we aren't returning an error, but I probably wouldn't kmyth_log any debug message for this.  Also, your if(!BIO_free) has an extra tab.